FV3/atmos_cubed_sphere/tools/coarse_grained_diagnostics.F90

```diff
@@ -0,0 +1,138 @@
+  use fv_arrays_mod, only: fv_atmos_type, fv_coarse_diag_type, fv_coarse_graining_type, fv_grid_bounds_type
+  use mpp_domains_mod, only: domain2d
+  use mpp_mod, only: FATAL, mpp_error
+  use coarse_graining_mod, only: block_sum, get_fine_array_bounds, get_coarse_array_bounds, MODEL_LEVEL, weighted_block_average
+  use time_manager_mod, only: time_type
+  
+  implicit none
+  private
+  
+  public :: fv_coarse_diag_init, fv_coarse_diag
+
+  integer :: tile_count = 1  ! Following fv_diagnostics.F90
+
+contains
+  
+  subroutine fv_coarse_diag_init(bd, Time, id_pfull, id_phalf, coarse_graining)
+    type(fv_grid_bounds_type), intent(in) :: bd
+    type(time_type), intent(in) :: Time
+    integer, intent(in) :: id_pfull, id_phalf
+    type(fv_coarse_graining_type), intent(inout) :: coarse_graining
+
+    integer :: is, ie, js, je, is_coarse, ie_coarse, js_coarse, je_coarse
+
+    call get_fine_array_bounds(bd, is, ie, js, je)
+    call get_coarse_array_bounds(coarse_graining%bd, is_coarse, ie_coarse, js_coarse, je_coarse)    
+    call initialize_coarse_diagnostic_axes(coarse_graining%domain, coarse_graining%nx_coarse, &
+         coarse_graining%id_x_coarse, coarse_graining%id_y_coarse, coarse_graining%id_xt_coarse, &
+         coarse_graining%id_yt_coarse)
+    
+    coarse_graining%id_pfull = id_pfull
+    coarse_graining%id_phalf = id_phalf
+
+    call register_coarse_diagnostics(coarse_graining%idiag, Time, &
+         coarse_graining%id_xt_coarse, coarse_graining%id_yt_coarse, id_pfull)
+  end subroutine fv_coarse_diag_init
+
+  subroutine initialize_coarse_diagnostic_axes(coarse_domain, &
+    nx_coarse, id_x_coarse, id_y_coarse, id_xt_coarse, id_yt_coarse)
+    type(domain2d), intent(in) :: coarse_domain
+    integer, intent(in) :: nx_coarse
+    integer, intent(inout) :: id_x_coarse, id_y_coarse, id_xt_coarse, id_yt_coarse
+
+    integer :: i, j
+    real, allocatable :: grid_x_coarse(:), grid_y_coarse(:), grid_xt_coarse(:), grid_yt_coarse(:)
+
+    allocate(grid_x_coarse(nx_coarse + 1))
+    allocate(grid_y_coarse(nx_coarse + 1))
+    allocate(grid_xt_coarse(nx_coarse))
+    allocate(grid_yt_coarse(nx_coarse))
+
+    grid_x_coarse = (/ (i, i=1, nx_coarse + 1) /)
+    grid_y_coarse = (/ (j, j=1, nx_coarse + 1) /)
+    grid_xt_coarse = (/ (i, i=1, nx_coarse) /)
+    grid_yt_coarse = (/ (j, j=1, nx_coarse) /)
+    
+    id_x_coarse = diag_axis_init('grid_x_coarse', grid_x_coarse, &
+         'degrees_E', 'x', 'Corner longitude', set_name='coarse_grid', &
+         Domain2=coarse_domain, tile_count=tile_count)
+    id_y_coarse = diag_axis_init('grid_y_coarse', grid_y_coarse, &
+         'degrees_N', 'y', 'Corner latitude', set_name='coarse_grid', &
+         Domain2=coarse_domain, tile_count=tile_count)
+
+    id_xt_coarse = diag_axis_init('grid_xt_coarse', grid_xt_coarse, &
+         'degrees_E', 'x', 'T-cell longitude', set_name='coarse_grid', &
```

Comment:
I know this is what the model uses to describe the x/y coordinates elsewhere, but these aren't actually latitude and longitude, right?

Reply:
Indeed, that's a good point.  I'll update these with some more accurate units/descriptions.

Reply:
See 6cd0964 for an update here.